pkg/controller/controller.go

```diff
@@ -225,8 +227,9 @@ func (c *Controller) processItem(key string) error {
 	}
 
 	funcObj := obj.(*spec.Function)
+	fUID := funcObj.Metadata.UID
 
-	err = utils.EnsureK8sResources(ns, name, &funcObj.Spec, c.clientset)
+	err = utils.EnsureK8sResources(ns, name, fUID, &funcObj.Spec, c.clientset)
```

Comment:
I even thought about this during my refactor and then forgot to change it :(  We should just pass the full `&funcObj` to `EnsureK8sResources` rather than chopping it up and passing the pieces separately...